lib/henkei.rb

```diff
@@ -223,39 +223,49 @@ def self.java_path
 
   # Internal helper for calling to Tika library directly
   #
-  def self.client_read(type, data)
+  def self.client_read(type, data, options={})
     IO.popen tika_command(type), 'r+' do |io|
-      io.write data
-      io.close_write
-      io.read
+      begin
+        with_timeout(options[:timeout]) do
+          io.write data
+          io.close_write
+          io.read
+        end
+      ensure
+        io.close
```

Comment:
per Ruby documentation:
```
If a block is given, Ruby will run the command as a child connected to Ruby with a pipe. Ruby's end of the pipe will be passed as a parameter to the block. At the end of block, Ruby closes the pipe and sets $?. In this case IO.popen returns the value of the block.
```
No need to explicitly close the io when used in block form